### PR TITLE
GDP-1002 Zip SOS Times Series algorithm

### PR DESCRIPTION
**Pull request by duselmann**
revised to accommodate TimeseriesDataset construction:
gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/FeatureTimeSeriesAlgorithm.java

New signatures to reduce variable string-date manipulation (and URI):
gdp-core-processing/src/main/java/gov/usgs/cida/gdp/coreprocessing/analysis/timeseries/
TimeseriesDataset.java
SOSClient.java

Updated test to match new method signatures:
gdp-core-processing/src/main/java/gov/usgs/cida/gdp/coreprocessing/analysis/timeseries/FeatureTimeseriesStatisticsTest.java

Added test for Zip directories testing.
gdp-core-processing/src/test/java/gov/usgs/cida/gdp/wps/algorithm/ZipOutputStreamTests.java

### CODE REVIEW

File: gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/FeatureTimeSeriesAlgorithm.java [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	public void setFeatureCollection(FeatureCollection<SimpleFeatureType,SimpleFeature> featureCollection) {
+		this.featureCollection = featureCollection;
+	}
+
+    @LiteralDataInput(
+            identifier=GDPAlgorithmConstants.OBSERVED_PROPERTY_IDENTIFIER,
+            title=GDPAlgorithmConstants.OBSERVED_PROPERTY_TITLE,
+            abstrakt=GDPAlgorithmConstants.OBSERVED_PROPERTY_ABSTRACT)
+    public void setObservedProperty(String observedProperty) {
+        this.observedProperty = observedProperty;
```
Comment (jiwalker-usgs) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/FeatureTimeSeriesAlgorithm.java]: observedProperty and timeseriesVariableName should be the same thing, only need one input.

**Comment by jiwalker-usgs** [comment]
This looks pretty good, have you been able to test it at all locally?

**Comment by duselmann** [comment]
ya, no. again the ground water has pulled me in two directions. I am willing to test. Where are these algorithms exposed?

**Comment by jiwalker-usgs** [comment]
If you get the process wps module deployed you need to add this algorithm to the wps_config.xml configuration.  Then you can POST the proper xml to the WPS endpoint and it will run the processing and give you a response.  There are some example xml execute requests in the test resources that you can base it on.  As far as the dataset and features, they should be in the JIRA, if not Dave can give you the best ones to test with.

**Comment by duselmann** [comment]
I could use the parameters you used to test your portion.

**Comment by duselmann** [comment]
I added an integration test, sample post XML metadata, and minor revisions for testing to work or discovered through testing.

**Comment by jiwalker-usgs** [comment]
Is this ready to be merged?

**Comment by duselmann** [comment]
@jiwalker-usgs  yes, It is ready for merge. I added a few test files. The significant code changes are in this diff if you would like to look at it prior to merge. https://github.com/USGS-CIDA/geo-data-portal/pull/210/commits/d9d1e32aa7058ae37c1d373685b47c4d2e55849e

Otherwise, I would say "let's merge"

**Comment by dblodgett-usgs** [comment]
I got the configuration up on dev. Seeing the algorithm as v 1.1.0 -- probably not a real problem, but it should probably start at 1.0.0?

**Pull request merged**